File: clients/python/coyote/apis/health.py
# This file is @generated

from .common import ApiBase
from ..models import (
    PingOut,
)


class HealthAsync(ApiBase):
    async def ping(
        self,
    ) -> PingOut:
        """Verify the server is up and running."""
        response = await self._request_asyncio(
            method="get", path="/api/v1/health/ping", path_params={}
        )
        return PingOut.model_validate(response.json())


class Health(ApiBase):
    def ping(
        self,
    ) -> PingOut:
        """Verify the server is up and running."""
        response = self._request_sync(
            method="get", path="/api/v1/health/ping", path_params={}
        )
        return PingOut.model_validate(response.json())

File: clients/javascript/src/apis/health.ts
// this file is @generated

import {
    type PingOut,
    PingOutSerializer,
} from '../models/pingOut';
import { HttpMethod, DiomRequest, type DiomRequestContext } from "../request";

export class Health {
    public constructor(private readonly requestCtx: DiomRequestContext) {}

    /** Verify the server is up and running. */
    public ping(
        ): Promise<PingOut> {
        const request = new DiomRequest(HttpMethod.GET, "/api/v1/health/ping");

        
        return request.send(
            this.requestCtx,
            PingOutSerializer._fromJsonObject,
        );
    }
}

